### PR TITLE
allow whitespace in expressions not wrapped in parenthesis

### PR DESCRIPTION
**Pull request by mlrawlings**
This change allows whitespace in complex expressions such as the following:

``` xml
<assign name=lastName.toUpperCase() + ', ' + firstName.toLowerCase()/>
```

While I think in many cases (especially when involving multiple attributes), the code would be easier to read if the developer were to still wrap more complex expressions in parenthesis, it shouldn't be necessary in all cases, and the above example is very clear and arguably 
more readable without the wrapping parenthesis.

Operators that begin with `>` would still need to be wrapped - in fact they already need to be wrapped even if there is no whitespace:

``` xml
<test x=(1>2)/>
```

This is accomplished by looking ahead for binary operators.  There is a side-effect of this change: anything that is listed in the `operators.js`, essentially becomes a reserved word and cannot be used as an attribute name.  For symbolic operators, an attribute cannot even start with the operator.  So, in the case of `<test a=1 +1=true/>`, the attribute name `+1` is **not** valid (in this context).

I think this is a fair trade-off and likely wouldn't affect anyone, but let's open this up for discussion as it could potentially break existing code.  **Is anyone using attributes that don't follow the rules for javascript variable names (aside from dashes in-between words)?**  If so, why and what for?

With the way things currently stand with this pull request, `<test +1=true/>` is actually still valid, if we decide to go this route, we should probably enforce the attribute name rule explicitly for all cases, rather than them sometimes working (if they're the first attribute, or follow an attribute without a value, or follow an argument).

### CODE REVIEW

**Comment by seangates** [comment]
For me, I actually like the enforcement of non-whitespace for each attribute. To me it makes it easy to distinguish between each attribute of a tag. Otherwise, I feel like I'd get caught up in trying to look for the next attribute, and working my way backward to find the end.

So, to me this is more readable in the template:
```
<assign name=(lastName.toUpperCase() + ', ' + firstName.toLowerCase()) age-in-years=(daysOld / 365)/>
```

Thinking about this more, though, is the idea of having semicolons to end statements, just as is done in JavaScript to signify a completed statement. What about something like this?:
```
<assign name=lastName.toUpperCase() + ', ' + firstName.toLowerCase(); age-in-years=daysOld / 365;/>
```

TL;DR I think there needs to be some distinguishing characteristics of a completed statement.
